### ISSUE TITLE
`datepicker`: "Maximum call stack size exceeded`

### ISSUE THREAD

**Issue by hognevevle** [formkit/formkit#1447]
### Reproduction

https://formkit.link/b7647b1d60174c0788f2f3da4b1fedc7

### Describe the bug

An infinite loop is triggered when advancing to the year part of the date picker.

### Environment

• OS: Windows
• Browser: Chrome
• Version: 126.0.6478.127

**Comment by justin-schroeder** [comment]
Hmm. perhaps this is a windows thing? On chrome on macOS im unable to duplicate it.

**Comment by hognevevle** [comment]
@justin-schroeder Looks like it. Doesn't reproduce on my Mac, not even in Windows running under Parallels. But on my "real" Windows-PC it reproduces in both Chrome, Edge and Firefox.

**Comment by TarSen99** [comment]
Having same issue on Mac OS. Any updates?

**Comment by kevcomparadise** [comment]
@justin-schroeder same issue here
Reproduction : [https://formkit.link/40ab8b24282190ae90c0a49b490ee818](https://formkit.link/40ab8b24282190ae90c0a49b490ee818)

**Comment by kevcomparadise** [comment]
> Hmm, still cannot duplicate this issue. @TarSen99, @kevcomparadise, @hognevevle — using @kevcomparadise reproduction (https://formkit.link/40ab8b24282190ae90c0a49b490ee818) can someone confirm these details:
> 
> 
> 
> - Your browser (and OS)
> 
> - Your timezone
> 
> - Your locale
> 
> 
> 
> 

Browser Chrome
Timezone GTM+1
Locale : FR

**Comment by sashamilenkovic** [comment]
@kevcomparadise Yeah, seems to be be throwing the error in certain time zones. Will take a look at this.

**Comment by sashamilenkovic** [comment]
@hognevevle @kevcomparadise @TarSen99 Update: an issue for this has been filed at @formkit/tempo https://github.com/formkit/tempo/issues/67.
